### PR TITLE
detect tty in kbc send

### PR DESCRIPTION
**Pull request by songgao**
r? @patrickxb @maxtaco Thanks!

(@zanderz helped me make sure the tty detection works on Windows and it behaves the same)

### CODE REVIEW

File: go/client/cmd_chat_send.go [hunk 1]
```diff
@@ -138,7 +139,7 @@ func (c *cmdChatSend) ParseArgv(ctx *cli.Context) (err error) {
 	c.setTopicName = ctx.String("set-topic-name")
 	c.setHeadline = ctx.String("set-headline")
 	c.clearHeadline = ctx.Bool("clear-headline")
-	c.useStdin = ctx.Bool("stdin")
+	c.useStdin = !isatty.IsTerminal(os.Stdin.Fd())
```
Comment (songgao) [inline go/client/cmd_chat_send.go]: Couldn't use the `TerminalUI` here since it's not initialized yet ...

Comment (maxtaco) [inline go/client/cmd_chat_send.go]: can you change the name of this member field?

Comment (maxtaco) [inline go/client/cmd_chat_send.go]: Also, the terminal UI stuff below is never used?!?!

Comment (maxtaco) [inline go/client/cmd_chat_send.go]: Ok, in that case just nuke the changes to GetTerminalUI then if they don't help

Comment (songgao) [inline go/client/cmd_chat_send.go]: OK; sounds good

**Comment by songgao** [comment]
@patrickxb sounds good; Will use the 4000 limit from `chat/msgchecker`. Although I'm curious how is it dangerous?

**Comment by maxtaco** [comment]
@songgao does `keybase chat read` also need to change?

**Comment by songgao** [comment]
@maxtaco I don't think so since it doesn't read anything from `stdin`. So even when pipe in, it'd still open TTY when needed

**Comment by maxtaco** [comment]
But what about piping standard output?  In that case (if standard output is not a TTY), you don't want to try to use a menu.  So I think `keybase chat read` needs to change too.

**Comment by songgao** [comment]
hmm Good point!

**Comment by maxtaco** [comment]
And will the resolver fail if it needs to resolve but we are not in interactive mode?

**Comment by songgao** [comment]
@maxtaco Yup

```
songgao at Song-ThinkPad in ~GOPATH/src/github.com/keybase/client/go/keybase (songgao/CORE-4122)
$ ./keybase chat read | cat
▶ WARNING Running in devel mode
▶ ERROR resolving conversation error: multiple conversations is found


songgao at Song-ThinkPad in ~GOPATH/src/github.com/keybase/client/go/keybase (songgao/CORE-4122)
$ echo hi | ./keybase chat send
Error parsing command line arguments: need exactly 1 argument to send from stdin

```

**Comment by maxtaco** [comment]
Ok great I am glad this was so easy. I am a :+1: after Patrick signs off. Thanks Song!!

**Comment by songgao** [comment]
Unfortunately this breaks the CLI tests since the stdin is not a TTY. Is it OK to ignore stdin if user has supplied a message through CLI? 

That is:

`echo foo | kbc send t_alice bar` would send bar instead of errors

**Comment by maxtaco** [comment]
@songgao I don't understand why this change is needed.  Is the CI test broken?  Why would you ever provide a non-TTY standard input and also a message at the same time?  Which test?

**Comment by maxtaco** [comment]
I see, this should be OK.

**Comment by songgao** [comment]
@maxtaco yeah the docker test likely opened stdin somehow, which makes `kbc send` think it's trying to pipe the stdin. Failed test: https://ci.keyba.se/job/kbfs-server/job/master/2730/execution/node/174/log/?consoleFull

**Comment by maxtaco** [comment]
OK, cool, thanks, 👍

**Comment by songgao** [comment]
Thanks Max!

**Comment by patrickxb** [comment]
@songgao if you pipe in a 100GB file, ioutil.ReadAll will read until out of memory and crash.  And if 4,000 character limit, might as well just read 4,000 characters.

**Comment by songgao** [comment]
@patrickxb got you; thanks! The `LimitReader`'s in the PR. 👍 to ⛵️ ?

**Pull request merged**